scrypto/src/crypto/ecdsa.rs

```diff
@@ -161,21 +176,23 @@ impl TryFrom<&[u8]> for EcdsaPrivateKey {
     type Error = ParseEcdsaPrivateKeyError;
 
     fn try_from(slice: &[u8]) -> Result<Self, Self::Error> {
-        if slice.len() == ECDSA_PRIVATE_KEY_LENGTH {
-            Ok(Self(copy_u8_array(slice)))
-        } else {
-            Err(ParseEcdsaPrivateKeyError::InvalidLength(slice.len()))
+        if slice.len() != EcdsaPrivateKey::LENGTH {
+            return Err(ParseEcdsaPrivateKeyError::InvalidLength(slice.len()));
         }
+        let sk =
+            SecretKey::from_be_bytes(slice).map_err(|_| ParseEcdsaPrivateKeyError::InvalidKey)?;
+        Ok(Self(sk))
     }
 }
 
 impl EcdsaPrivateKey {
     pub fn to_vec(&self) -> Vec<u8> {
-        self.0.to_vec()
+        self.0.to_be_bytes().as_slice().to_vec()
     }
 }
 
-// Private key is not a custom type, as we don't expect it to be passed around
+// Temporarily for simulator
```

Comment:
Why needed? Should have larger alarm bells in comment such as `FIXME`

Reply:
The simulator needs a way to store the private key and it's using sbor.

I agree that it doesn't have to be in the custom type list, but I don't think this should be FIXME. It would be nice to have our library not encouraging clients to store private key. It's ultimately their responsibility to keep private key safely (they will find an alternative way to encode if sbor doesn't support it).

I will see if I can remove it.

Reply:
I think private key serialization/deserialization would flag any auditor immediately (we were flagged for this before). And from security perspective, we want to not have any private key handling in radix-engine\scrypto since we want to ensure no logic ever requires storing or signing with private keys.

Reply:
Feel like the auditor is being lazy - don't want to check how they're being used. 

All crypto library I've seen supports `SEC1 ASN.1 DER` encoding and decoding.